guidelines/component-usage.md

```diff
@@ -1 +1,106 @@
+## Table of content
 
+1. [Syntax](#syntax)
+2. [Formatting](#formatting)
+3. [Component Structure](#component-structure)
+4. [Eslint](#eslint)
+
+
+## Syntax
+
+1. #### Naming Style **[⬆](#table-of-content)**
+
+   * Always use `.jsx` | `.tsx` extensions.
+   * Use UpperCamelCase as a components name.
+   * Use lowerCamelCase for prop names.
+   * It's recommended to use suffixes `Component` or `Container` for explicit definition of "dumb" and "smart" components: `UserProfileComponent` and `UserProfileContainer`.
+   * Component name should be the same as the file name. This means that the use of `index` files is forbidden for components. It's easier to find components by filename rather than by folder name.
+   * Component filenames should be in UpperCamelCase. It's allowed to use dots (`.`) to explicitly mention type of component: `Dashboard.component.tsx`, `UserProfile.container.tsx`.
+   * Props and state interface definitions shouldn't be prefixed with `I`. Use UpperCamelCase instead.
+   * Exported props types should have a suffix `Props`: `ProgressBarProps`. Not exported props and state interfaces recommended naming `Props` and `State` respectively.   
+
+## Formatting
```

Comment:
I think most of this section should be enforced by an eslintrc file, not be specified as guidelines.  (Edit: I see monkey mentioned this on IRC already.)

Reply:
Thanks! Fixed